woocommerce/woocommerce
# Marketing: Add "Sales" column in Campaigns card

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/43248.

In this PR, we add a new Sales column in the Campaigns card in the Marketing page.

When there are no values for the cost or sales columns, it will display a dash "-".

The sales column implementation is based on the existing implementation for cost column.

Screenshot:

![image](https://github.com/woocommerce/woocommerce/assets/417342/6c2d1ddb-83ff-4417-92c9-bfefe7b99a4c)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Note: to test the changes, you may need to use some test data in some marketing channel extensions. The following is an example of how to do that using Google Listings and Ads:

1. Modify the line https://github.com/woocommerce/google-listings-and-ads/blob/72543805c6a24191c33c5cd0177d66b07537b18b/src/MultichannelMarketing/GLAChannel.php#L75 to the following:

```php
		if ( apply_filters( 'woocommerce_gla_enable_mcm', true ) === true ) {
```

2. Modify the `get_campaigns` function in https://github.com/woocommerce/google-listings-and-ads/blob/72543805c6a24191c33c5cd0177d66b07537b18b/src/MultichannelMarketing/GLAChannel.php#L178-L206, so that it returns some sample marketing campaigns from MailPoet and Google Listings and Ads:

```php
	public function get_campaigns(): array {
		return [
			new MarketingCampaign(
				'1',
				$this->campaign_types['google-ads'],
				'Campaign 1 - no cost, no sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
			),
			new MarketingCampaign(
				'2',
				$this->campaign_types['google-ads'],
				'Campaign 2 - no cost, with sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				null,
				new Price( '30', 'USD' ),
			),
			new MarketingCampaign(
				'3',
				$this->campaign_types['google-ads'],
				'Campaign 3 - with cost, no sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '30', 'USD' ),
			),
			new MarketingCampaign(
				'4',
				$this->campaign_types['google-ads'],
				'Campaign 4 - with cost, with sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '5,000', 'USD' ),
				new Price( '6,000', 'USD' ),
			),
			new MarketingCampaign(
				'5',
				$this->campaign_types['google-ads'],
				'Campaign 5',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '5,000', 'USD' ),
				new Price( '6,000', 'USD' ),
			),
			new MarketingCampaign(
				'6',
				$this->campaign_types['google-ads'],
				'Campaign 6',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '6,888.5', 'USD' ),
			),
		];
	}
```

Test Steps:

1. Go to the WooCommerce > Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`. 
2. You should see Campaigns card like the screenshot above.
    - You should see Cost and Sales column.
    - If there are no values for cost or sales, a dash "-" would be displayed.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
